.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,27 @@
+name: Docker notebook tests
+
+on:
+  push:
+    [ main ]
+    
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build the containers
+        run: docker compose -f docker-compose-dev.yml --profile jupyter build
+      - name: Run the jupyter profile
+        run: docker compose -f docker-compose-dev.yml --profile jupyter up -d
+      - name: Give containers time to initialize
+        shell: bash
+        run: sleep 90
+      - name: Test notebooks in the docker environment
+        shell: bash
+        run: docker exec qs-jupyter "bash" "-c" "pip install nbmake pytest && pytest --nbmake --ignore=/home/jovyan/serverless/guides/07_working_with_datasets.ipynb --ignore=/home/jovyan/serverless/examples/06_electronic_structure_problem.ipynb /home/jovyan/serverless/"
```

Comment:
We will ignore two tests here.

#671 PR will handle the ESProblem notebook
#672 explains why we will ignore the dataset notebook here to start. I will follow up with @IceKhan13 about why Ray is behaving like this in CI and address this issue